File: resolver/parser.go
package resolver

import (
	"github.com/miekg/dns"
	//"log"
	"net"
)

func Parser(message *dns.Msg) *dns.A {
	var ip string
	for _, msg := range message.Question {
		switch msg.Qclass {
		case dns.TypeA:
			ip = Query(message)
		}
	}

	return &dns.A{
		Hdr: dns.RR_Header{Name: message.Question[0].Name, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
		A:   net.ParseIP(ip),
	}
}

File: Python Server/Resolver/parser.py
import dns.message
import dns.rdatatype
from Resolver.query import Query 

def Parser(data: dns.message.Message) -> dns.message.Message:
    ip = dns.message.Message()
    for q in data.question:
        if q.rdtype == dns.rdatatype.A: #https://dnspython.readthedocs.io/en/latest/rdatatype-list.html
            ip = Query(q.name)
    return ip
